scenariorun/src/test/kotlin/com/cosmotech/scenariorun/azure/ScenarioRunServiceImplTests.kt

```diff
@@ -70,14 +71,16 @@ class ScenarioRunServiceImplTests {
   private lateinit var eventPublisher: CsmEventPublisher
 
   @MockK(relaxed = true) private lateinit var workflowService: WorkflowService
+  @MockK(relaxed = true) private lateinit var azureDataExplorerClient: AzureDataExplorerClient
```

Comment:
Please consider adding a unit test case, to ensure that your new `deleteDataFromScenarioRunId` method on `AzureDataExplorerClient` really does get called when deleting a ScenarioRun.
This is a method call verification test. See https://mockk.io/#verification-atleast-atmost-or-exactly-times